extension/sumologicextension/extension.go

```diff
@@ -556,17 +556,17 @@ func (se *SumologicExtension) registerCollectorWithBackoff(ctx context.Context,
 	}
 }
 
-func (se *SumologicExtension) heartbeatLoop() {
+func (se *SumologicExtension) heartbeatLoop(ctx context.Context) {
 	if se.registrationInfo.CollectorCredentialID == "" || se.registrationInfo.CollectorCredentialKey == "" {
 		se.logger.Error("Collector not registered, cannot send heartbeat")
 		return
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
thanks for sharing this. I wanted the long running operation (heartbeat) to be cancelled immediately on calling shutdown (hence the parent context). The reason for this is because I think there's some delay before the heartbeats are cancelled on component shutdown, that's causing this flaky test issue

Reply:
Having said that perhaps this isn't the best way to do this. Maybe I could add a slight delay in the test instead.

Reply:
this PR- https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32937 removes the flaky test.